k4FWCore/components/MetadataSvc.cpp

```diff
@@ -38,14 +38,13 @@ StatusCode MetadataSvc::initialize() {
     error() << "Unable to locate the EventDataSvc" << endmsg;
     return StatusCode::FAILURE;
   }
-
-  m_frame = std::make_unique<podio::Frame>();
-
   return StatusCode::SUCCESS;
 }
 
 StatusCode MetadataSvc::finalize() { return Service::finalize(); }
 
-void MetadataSvc::setFrame(podio::Frame&& fr) { m_frame = std::make_unique<podio::Frame>(std::move(fr)); }
+const podio::Frame* MetadataSvc::getFrame() const { return m_frame.get(); }
```

Comment:
Make it private and the IOSvc a friend of the MetadataSvc or something to hide from other users.

Reply:
Moved `getFrame` in `IMetadataSvc` (interface) to protected and made friends with `Writer`